sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/docker_client.py

```diff
@@ -189,6 +192,7 @@ def create_deployment(
         module_logger.debug("Mounting volumes: '%s'\n", volumes)
         module_logger.debug("Setting environment variables: '%s'\n", environment)
         container_name = _get_container_name(endpoint_name, deployment_name)
+        device_requests = [docker.types.DeviceRequest(capabilities=[["gpu"]])] if local_enable_gpu else None
```

Comment:
I believe this also requires the `nvidia-docker` package. What happens if that is not installed?

Reply:
I tried with minimal gpu image, it works as expected.
It doesn't work with minimal cpu image. (It doesn't give any error)
In my local system Nvidia driver and docker is installed, other than I have not installed anything. 
I think nvidia-docker package is not required.

Reply:
This is set on the host so minimal cpu image should also work. Can you check why?

Are you using WSL + Docker desktop? I do believe this setup does not require additional packages. Can you try to replicate the setup in a raw VM?

Reply:
I have updated the code, it will work on all images as it is host dependent.